student/src/main/java/tds/student/services/remote/RemoteAccommodationsService.java

```diff
@@ -27,55 +34,150 @@
+    List<Accommodation> assessmentAccommodationsResponse = assessmentRepository.findAccommodations(opportunityInstance.getExamClientName(), testKey);
+    
+    return mapExamAccommodationsToLegacyAccommodations(assessment, assessmentAccommodationsResponse, examAccommodationResponse, isGuestSession);
   }
   
+  /* This method is a port of AccommodationService.getTestSegments() between [398 - 426] with some logic located in getApproved()
+  *  The "Accommodations" class is what the legacy code will use to create the TDS-Student-Accs cookie */
+  private List<Accommodations> mapExamAccommodationsToLegacyAccommodations(final Assessment assessment, final List<Accommodation> assessmentAccommodations,
+                                                     final List<ExamAccommodation> approvedExamAccommodations, final boolean isGuestSession) {
+    // accCode -> ExamAccommodation
+    Map<String, ExamAccommodation> approvedExamAccommodationMap = new HashMap<>();
+    // segmentPosition -> legacy Accommodations object (per segment + assessment)
+    Map<Integer, Accommodations> segmentPositionToAccommodations = new HashMap<>();
+    Accommodations retAccommodations;
+    ExamAccommodation approvedExamAccommodation;
+    int segmentPosition;
+    
+    // Map the accommodation keys (code) to their exam accommodations for quick lookup
+    for (ExamAccommodation examAccommodation : approvedExamAccommodations) {
+      approvedExamAccommodationMap.put(examAccommodation.getCode(), examAccommodation);
+    }
+    
+    /* We need to filter only approve ExamAccommodations - although the Assessment acccommodations contain
+       the majority of the accommodation metadata  */
+    for (Accommodation accommodation : assessmentAccommodations) {
+      // Filter only approved accommodations
+      approvedExamAccommodation = approvedExamAccommodationMap.get(accommodation.getCode());
+    
+      if (approvedExamAccommodation != null && approvedExamAccommodation.getType().equalsIgnoreCase(accommodation.getType())) {
+        segmentPosition = accommodation.getSegmentPosition();
+        retAccommodations = segmentPositionToAccommodations.get(segmentPosition);
+      
+        // If there are no Accommodations created for this segment position, we need to create one.
+        if (retAccommodations == null) {
+          // Accommodation context is either the segmentId or assessmentId, depending on whether the accommodation is
+          // segment specific or global to the assessment
+          retAccommodations = new Accommodations(segmentPosition, accommodation.getContext(), getLabelForAssessmentOrSegment(segmentPosition, assessment));
+          segmentPositionToAccommodations.put(segmentPosition, retAccommodations);
+        }
+      
+        /* AccommodationService [399-400] - If this isn't a guest session, then no need to check isDisabledOnGuest flag */
+        if (!isGuestSession || (isGuestSession && !accommodation.isDisableOnGuestSession())) {
```

Comment:
the `isGuestSession` in the OR part is not needed since if it gets there it means it is a guest session